src/monitor/commands/edit-monitor-availability.command.ts

```diff
@@ -0,0 +1,138 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/database/prisma.service';
+import { days } from '../dto/monitor-availability.dto';
+import {
+  EditMonitorBodyMissingFieldsException,
+  EmptyAvailabilityException,
+  MonitoringNotFoundException,
+  UserLoggedNotResponsableForMonitoringException,
+} from '../utils/exceptions';
+import { EditMonitorAvailabilityDTO } from '../dto/edit-monitor-availability.dto';
+
+@Injectable()
+export class EditMonitorAvailabilityCommand {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async execute(
+    monitorId: number,
+    tokenUserId: number,
+    preferentialPlace: string,
+    availability: days[],
+  ): Promise<EditMonitorAvailabilityDTO> {
+    if (!preferentialPlace && !availability)
+      throw new EditMonitorBodyMissingFieldsException();
+
+    if (!availability.length) {
+      throw new EmptyAvailabilityException();
+    }
+
+    const monitoring = await this.prisma.monitor.findFirst({
+      where: {
+        id: monitorId,
+      },
+    });
+
+    if (!monitoring) throw new MonitoringNotFoundException();
+
+    if (monitoring.student_id != tokenUserId)
+      throw new UserLoggedNotResponsableForMonitoringException();
+
+    await this.prisma.$transaction(async (tx) => {
```

Comment:
Fica dando o erro `Transaction API error: Transaction already closed: Could not perform operation` e aí isso quebra a api. Depois disso não dá mais pra fazer request até reiniciar o container

Reply:
Pra evitar problema com isso, pode tirar o uso da transaction. Prisma é uma *****

Reply:
Amg. Dei uma pesquisada aqui e talvez tenha conseguido resolver. Se nao funcionar agora a gente abandona msm